File: courses/algo_struct/Parallellogram/src/std/Student.java
package std;

public class Student {
	private String name;
	private int group;
	private float mark;

	public Student(String name, int group) {
		this.name = name;
		this.group = group;
		this.mark = 0.0F;
	}

	public void output() {
		System.out.println("Name : " + name + "\nGroup: " + group + "\nMark : " + mark);
	}

	public void setMark(float mark) {
		this.mark = mark;
	}
}

File: courses/prog_base_2/tests/test_2/test_server/test_server/student.c
#include "student.h"
#include "http_server.h"
#include <libxml\xmlwriter.h>
#include <libxml\encoding.h>
#include <string.h>
student student_new(char * name, char * group_name, size_t variant)
{
	student st;
	strcpy(st.name, name);
	strcpy(st.group_name, group_name);
	st.variant = variant;
	return st;
}

char * student_to_xml(student * student)
{
	char * output = malloc(BUFFER_LENGTH);
	char write_buffer[BUFFER_LENGTH];
	xmlTextWriterPtr writer;
	xmlBufferPtr xml_buf;
	xml_buf = xmlBufferCreate();
	writer = xmlNewTextWriterMemory(xml_buf, 0);
	xmlTextWriterStartDocument(writer, "1.0", "ASCII", NULL);
	xmlTextWriterStartElement(writer, "student");
	xmlTextWriterWriteElement(writer, "name", student->name);
	xmlTextWriterWriteElement(writer, "group", student->group_name);
	char buff[100];
	xmlTextWriterWriteElement(writer, "variant", itoa(student->variant, buff, 10));
	xmlTextWriterEndElement(writer);
	xmlTextWriterEndDocument(writer);
	xmlFreeTextWriter(writer);
	strcpy(output, (const char *)(xml_buf->content));
	return output;
}
